### PR TITLE
Kotlin/data classes

### PR DESCRIPTION
**Pull request by fourlastor**
This PR converts our data classes from AutoValue to kotlin data classes.

I didn't touch NearIT related classes because they're most likely to change soon

The current state isn't pretty, the usage of `.Companion.create` everywhere and the return of `get*` methods doesn't make me happy, but this is step 1, I'll convert the other classes a bit at a time.

### CODE REVIEW

File: .idea/inspectionProfiles/Squanchy.xml [hunk 1]
```diff
@@ -136,7 +136,7 @@
     <inspection_tool class="Guava" enabled="false" level="WARNING" enabled_by_default="false" />
     <inspection_tool class="GwtUiFieldErrors" enabled="false" level="ERROR" enabled_by_default="false" />
     <inspection_tool class="GwtUiHandlerErrors" enabled="false" level="ERROR" enabled_by_default="false" />
-    <inspection_tool class="IfMayBeConditional" enabled="true" level="INFO" enabled_by_default="true" />
```
Comment (rock3r) [inline .idea/inspectionProfiles/Squanchy.xml:139]: YES! 💯

File: app/src/main/java/net/squanchy/about/licenses/Libraries.kt [hunk 2]
```diff
@@ -0,0 +1,31 @@
+package net.squanchy.about.licenses
+
+internal object Libraries {
+
+    val LIBRARIES = listOf(
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/about/licenses/Libraries.kt:5]: I suppose this should contain Kotlin too now :P

File: app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.java [hunk 3]
```diff
@@ -23,7 +23,7 @@
     private static final String EXTRA_EVENT_ID = EventDetailsActivity.class.getCanonicalName() + ".event_id";
     private static final int REQUEST_CODE_SIGNIN = 1235;
 
-    private final CompositeDisposable subscriptions = new CompositeDisposable();;
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/eventdetails/EventDetailsActivity.java:26]: Imma fix this in develop too

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.kt [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+        val speakers: List<Speaker>,
+        val experienceLevel: Optional<ExperienceLevel>,
+        val dayId: String,
+        val type: Type,
+        val favorited: Boolean,
+        val description: Optional<String>,
+        val timeZone: DateTimeZone
+) {
+
+    fun speakersNames(): String {
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/domain/view/Event.kt:26]: Do we still actually use this? I suppose this should be moved to the UI layer :P

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.kt [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    enum class Type constructor(private val rawType: String) {
+        REGISTRATION("registration"),
+        TALK("talk"),
+        KEYNOTE("keynote"),
+        COFFEE_BREAK("coffee_break"),
+        LUNCH("lunch"),
+        SOCIAL("social"),
+        OTHER("other");
+
+
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/schedule/domain/view/Event.kt]: One extra whiteline here, leave one only

File: app/src/main/java/net/squanchy/support/widget/SpeakerView.java [hunk 6]
```diff
@@ -74,7 +74,7 @@ public void updateWith(List<Speaker> speakers, Optional<OnSpeakerClickListener>
     }
 
     private String toCommaSeparatedNames(List<Speaker> speakers) {
-        return TextUtils.join(", ", map(speakers, Speaker::name));
+        return TextUtils.join(", ", map(speakers, Speaker::getName));
```
Comment (rock3r) [inline app/src/main/java/net/squanchy/support/widget/SpeakerView.java:77]: Aha! So the one on... was it `Event` is really not necessary anymore. Noice.

**Review by rock3r — COMMENTED** [review]
Few comments, will also wait for the changes to get the CI to be green before yay'ing

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.kt (hunk 4 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/schedule/domain/view/Event.kt:26]: I literally converted it, didn't move stuff

File: app/src/main/java/net/squanchy/about/licenses/Libraries.kt (hunk 2 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/about/licenses/Libraries.kt:5]: next PR

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.kt (hunk 4 above)
Comment (rock3r) [reply app/src/main/java/net/squanchy/schedule/domain/view/Event.kt:26]: Was mostly a note to self :)

File: app/src/main/java/net/squanchy/schedule/domain/view/Event.kt (hunk 5 above)
Comment (fourlastor) [reply app/src/main/java/net/squanchy/schedule/domain/view/Event.kt]: 6d8ac2b

**Review by rock3r — APPROVED** [review]

**Pull request merged**